lib/Tool.php

```diff
@@ -659,11 +659,11 @@ public static function interfaceExists($class)
 
     /**
      * @param string $class
-     * @param string $type
+     * @param string $type (e.g. 'class', 'interface', 'trait')
      *
      * @return bool
      */
-    protected static function classInterfaceExists($class, $type)
+    public static function classInterfaceExists($class, $type)
```

Comment:
Wouldn't it be better to leave this protected and add a facade method like for the other two options?

```php
    /**
     * @static
     *
     * @param string $trait
     *
     * @return bool
     */
    public static function traitExists($trait)
    {
        return self::classInterfaceExists($trait, 'trait');
    }
```

Reply:
don't see a reason why it can't be public

Reply:
Maybe that's right. 

It is just strange, because `$type` can only take 3 values and there are explicit public methods for the other 2.

So why not add a third one, that makes the intention clear, instead of forcing the user to call a method named `classInterfaceExists` to check if a **trait** exists (and that's the only real reason to call this method because for the other cases there are simpler, better named methods).